meshio/vtu/_vtu.py

```diff
@@ -43,16 +43,93 @@ def _cells_from_data(connectivity, offsets, types, cell_data_raw):
         [[0], numpy.where(types[:-1] != types[1:])[0] + 1, [len(types)]]
     )
 
+    # boolean needed for polyhedron cells
+    found_polyhedron_faces_before = False
+
     cells = []
     cell_data = {}
+    polyhedron_faces = {}
     for start, end in zip(b[:-1], b[1:]):
         try:
             meshio_type = vtk_to_meshio_type[types[start]]
         except KeyError:
             raise ReadError("File contains cells that meshio cannot handle.")
-        if meshio_type == "polygon" or meshio_type == "polyhedron":
+        if meshio_type in ["polygon", "polyhedron"]:
             # Polygons and polyhedra have unknown and varying number of nodes per cell.
 
+            if meshio_type == "polyhedron":
+                # The data format for face-cells is:
+                # num_faces_cell_0,
+                #   num_nodes_face_0, node_ind_0, node_ind_1, ..
+                #   num_nodes_face_1, node_ind_0, node_ind_1, ..
+                #   ...
+                # num_faces_cell_1,
+                #   ...
+                # See https://vtk.org/Wiki/VTK/Polyhedron_Support for more.
+
+                # There is a tacit assumption here that the cell-data keys 'faces'
+                # and 'faceoffsets' are reserved for this purpose.
```

Comment:
Wait, polyhedra are not part of the standard?

Reply:
It is a part of the vtu format and has been for (at least) a few years. However, as in this PR, the 'anything goes'-property of polyhedra calls for a specialized data forma

Reply:
I cannot find the word "polyhedron" in [the spec](https://vtk.org/wp-content/uploads/2015/04/file-formats.pdf).

Reply:
The link you posted is about the VTK software package, not the file formats, isn't it?

Reply:
Yes, I believe so

Reply:
Okay. And did I understand correctly that `faces` and `faceoffsets` are cell data (connected to which cell?) that contain polyhedral info?

Reply:
Yes. The data for cells are merged, so that `faces` contains datafor all polyhedral cells, `faceoffsets` gives the end of the data for each cell.

Reply:
I don't get it. Can we have a video call perhaps? I think we can better talk about the two PRs that way. Can you get back to me at nico.schloemer@gmail.com?